### PR TITLE
Updated Matter Python REPL

### PR DESCRIPTION
**Pull request by mrjerryjohns**
This resurrects the chip-repl to be more usable and user-friendly, especially around how pretty printing, help text/documentation are generated within iPython.

Notably, it leverages the 'rich' module to:
    - Provide capabilities to inspect the methods/values in any given
      class/module.
    - Pretty printing of cluster objects.

Some notable screen-shots are below.

The new `matterhelp()` command:
Screen Shot 2021-11-08 at 3 20 45 PM

Pretty-printing cluster objects, including nested structs/lists with correct typing information:

Screen Shot 2021-11-08 at 3 23 36 PM

Inspecting cluster objects to understand the types of their various fields:
Screen Shot 2021-11-08 at 3 24 57 PM

### CODE REVIEW

**Review by tcarmelveilleux — APPROVED** [review]

**Review by erjiaqing — APPROVED** [review]
Awesome

**Review by yunhanw-google — APPROVED** [review]

**Comment by yunhanw-google** [comment]
Document update?

ninja: Entering directory `./out/debug'
ninja: error: '../../src/controller/python/ChipReplStartup.py', needed by 'linux_x64_gcc/controller/python/chip-0.0-cp37-abi3-linux_x86_64.whl', missing and no known rule to make it

See this error, you miss this file ChipReplStartup.py? forget to add it?

File: src/controller/python/BUILD.gn [hunk 1]
```diff
@@ -95,6 +95,7 @@ pw_python_action("python") {
     {
       src_dir = "."
       sources = [
+        "ChipReplStartup.py",
```
Comment (andy31415) [inline src/controller/python/BUILD.gn:98]: add this file to the PR

**Review by Damian-Nordic — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/controller/python/ChipReplStartup.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    coloredlogs.install(level='DEBUG')
+    chip.logging.RedirectToPythonLogging()
+    logging.getLogger().setLevel(logging.ERROR)
+
+
+def matterhelp(classOrObj=None):
+    if (classOrObj == None):
+        inspect(builtins.devCtrl, methods=True, help=True, private=False)
+    else:
+        inspect(classOrObj, methods=True, help=True, private=False)
```
Comment (electrocucaracha) [inline src/controller/python/ChipReplStartup.py:47]: Shouldn't be better to use its falsy value?
```python
>>> bool(None)
False
```
```suggestion
    if classOrObj:
        inspect(classOrObj, methods=True, help=True, private=False)
    else:
        inspect(builtins.devCtrl, methods=True, help=True, private=False)
```

Comment (mrjerryjohns) [reply src/controller/python/ChipReplStartup.py:47]: In the interest of getting this in, I'll tackle this as a clean-up later!

**Pull request merged**